
```{r}
rm(list=ls())
library(readr)
library(glue)
# install.packages("tidyverse")
library(tidyverse)
# install.packages("ggplot2")
library(ggplot2)
library(googlesheets4)
# install.packages("RColorBrewer")
library(RColorBrewer)
# install.packages("plotly")
library(plotly)
install.packages("ggpubr")
library(ggpubr)

```




```{r}
csv_file = "./somatic_alterations_from_tumors.csv"
data <- read_csv(csv_file)
# print unique values of colnames
print(unique(colnames(data)))
# print unique values of each column (limit to 20 values per column)
# for (col in colnames(data)) {
#   print(paste("Column:", col))
#   print(data[[col]] %>% distinct() %>% head(20))
}
# data

```

```{r}

# Load the CSV file
df <- read.csv("../somatic_alterations_from_tumors.csv", stringsAsFactors = FALSE)

# Loop through each column
# for (col_name in names(df)) {
#   cat("Column:", col_name, "\n")
#   
#   # Get unique values (excluding NA)
#   unique_vals <- unique(na.omit(df[[col_name]]))
#   num_unique <- length(unique_vals)
#   
#   cat("Number of unique values:", num_unique, "\n")
#   
#   # Print up to 30 values
#   print(head(unique_vals, 30))
#   
#   cat("-----\n")
# }




```

```{r}

top_genes <- names(sort(table(df$Hugo_Symbol), decreasing = TRUE))[1:50]
df_top <- df[df$Hugo_Symbol %in% top_genes, ]
colnames(df_top)


# טען ggplot2 אם עוד לא טענת
library(ggplot2)

# צייר scatter plot
ggplot(df_top, aes(x = Hugo_Symbol, y = Diagnosis.Age, color = Variant_Type)) +
  scale_x_discrete(expand = expansion(mult = c(0.01, 0.01))) + 
  geom_point(position = position_jitterdodge(jitter.width =  0.4), alpha=0.5) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(title = "Scatter Plot of Gene vs Age", x = "Gene", y = "Age")

```



```{r}
gene_number <- 20


# top_stages <- names(sort(table(df$Tumor.Stage), decreasing = TRUE))[1:4]
top_cancers <- names(sort(table(df$Cancer.Type), decreasing = TRUE))[1:3]

# df_top <- df[df$Hugo_Symbol %in% top_genes, ]
# df_top <- df_top[df_top$Tumor.Stage %in% top_stages, ]
df_top <- df[df$Cancer.Type %in% top_cancers, ]

df_grouped <- df_top %>%
  group_by(Hugo_Symbol, Diagnosis.Age, Variant_Type, Cancer.Type) %>%
  summarise(count = n(), .groups = "drop")
df_grouped <- df_grouped %>%
  filter(Variant_Type %in% c("SNP", "DEL", "INS"))

df_grouped <- df_grouped %>% 
  mutate(Variant_Type = fct_reorder(Variant_Type, count, .fun = sum, .desc = TRUE))

top_genes <- names(sort(table(df$Hugo_Symbol), decreasing = TRUE))[1:gene_number]
df_grouped <- df_grouped[df_grouped$Hugo_Symbol %in% top_genes, ]


# df_grouped$Hugo_Symbol <- factor(df_grouped$Hugo_Symbol, levels = top_genes)

ggplot(df_grouped, aes(x = Hugo_Symbol, y = Diagnosis.Age, size = count, color = Cancer.Type)) +
  scale_size_continuous(range = c(1, 10)) +
  geom_jitter(width = 0.3, alpha = 0.7) +  
  theme_minimal() +
  scale_color_brewer(palette = "Set1") +
  # scale_color_manual(values = c("#377eb8", "#4daf4a", "#e41a1c")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 14)) +
  labs(title = glue("Mutated Genes by Age (Top {gene_number} genes)"),
       x = "Gene (Hugo_Symbol)", y = "Age", size = "Sample Count") +
  facet_wrap(~Variant_Type, nrow=3)
  # make the labs font bigger
  





```

```{r}

ggplot(df_grouped, aes(x = Hugo_Symbol, y = Diagnosis.Age, size = count, color = Variant_Type)) +
  scale_size_continuous(range = c(1, 10)) +
  geom_jitter(width = 0.3, alpha = 0.7) +  
  theme_minimal() +
  scale_color_brewer(palette = "Dark2") +
  scale_color_manual(values = c("#377eb8", "#4daf4a", "#e41a1c", "#984ea3", "#ff7f00")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 14)) +
  labs(title = glue("Mutated Genes by Age (Top {gene_number} genes)"),
       x = "Gene (Hugo_Symbol)", y = "Age", size = "Sample Count")


```

```{r}
# df <- read.csv("./somatic_alterations_from_tumors.csv", stringsAsFactors = FALSE)

df_smoke_data <- df %>% 
  filter(Smoke.Status != "Unknown", Variant_Type %in% c("SNP", "DEL", "INS")) %>%
  # filter(Cancer.Type %in% names(sort(table(df$Cancer.Type), decreasing = TRUE))[1:gene_number]) %>%
  group_by(Cancer.Type, Smoke.Status) %>%
  summarise(count = n(), .groups = "drop") 
# top_genes <- names(sort(table(df$Hugo_Symbol), decreasing = TRUE))[1:gene_number]
# df_grouped <- df_grouped[df_grouped$Hugo_Symbol %in% top_genes, ]

ggplot(df_smoke_data, aes(x = Cancer.Type, y = count, fill = Smoke.Status)) +
  geom_col(position = "dodge")

# print how many are smokers
# smokers_count <- nrow(df_smoke_data[df_smoke_data$Smoke.Status == "Smoker", ])
# non_smokers_count <- nrow(df_smoke_data[df_smoke_data$Smoke.Status == "Nonsmoker", ])

```






```{r}

gene_number <- 20

# 
# # top_stages <- names(sort(table(df$Tumor.Stage), decreasing = TRUE))[1:4]
# top_cancers <- names(sort(table(df$Cancer.Type), decreasing = TRUE))[1:3]
# 
# # df_top <- df[df$Hugo_Symbol %in% top_genes, ]
# # df_top <- df_top[df_top$Tumor.Stage %in% top_stages, ]
# df_top <- df[df$Cancer.Type %in% top_cancers, ]
# 
# df_grouped <- df_top %>%
#   group_by(Hugo_Symbol, Diagnosis.Age, Sm) %>%
#   summarise(count = n(), .groups = "drop")
# df_grouped <- df_grouped %>%
#   filter(Variant_Type %in% c("SNP", "DEL", "INS"))
# 
# df_grouped <- df_grouped %>% 
#   mutate(Variant_Type = fct_reorder(Variant_Type, count, .fun = sum, .desc = TRUE))





top_genes <- names(sort(table(df$Hugo_Symbol), decreasing = TRUE))[1:gene_number]
df_filtered <- df[df$Hugo_Symbol %in% top_genes, ]
df_filtered %>% 
  filter(Smoke.Status != "Unknown", Variant_Type %in% c("SNP", "DEL", "INS")) %>%
  group_by(Hugo_Symbol, Diagnosis.Age, Sex) %>%
  summarise(count = n(), .groups = "drop") -> df_filtered

df_filtered$Hugo_Symbol <- factor(df_filtered$Hugo_Symbol, levels = top_genes)

ggplot(df_filtered, aes(x = Hugo_Symbol, y = Diagnosis.Age, size = count)) +
  geom_boxplot(aes(fill = Sex), alpha = 0.5, outlier.shape = NA) +
  theme_minimal() +
  scale_color_brewer(palette = "Set1") +
  # scale_color_manual(values = c("#377eb8", "#4daf4a", "#e41a1c")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 14)) +
  labs(title = glue("Mutated Genes by Age (Top {gene_number} genes)"),
       x = "Gene (Hugo_Symbol)", y = "Age", size = "Sample Count") 
  # facet_wrap(~Variant_Type, nrow=3)
  # make the labs font bigger
  



```

```{r}
# Non significant correlation
# Filter for key genes of interest
key_genes <- c("TP53", "KRAS", "EGFR")

mutation_rate_by_smoking <- df %>%
  filter(Hugo_Symbol %in% key_genes & Smoke.Status != "Unknown") %>%
  group_by(Smoke.Status, Hugo_Symbol) %>%
  summarise(
    Num_Patients = n_distinct(PATIENT_ID),
    .groups = "drop"
  ) %>%
  group_by(Hugo_Symbol) %>%
  mutate(
    Percent = Num_Patients / sum(Num_Patients) * 100
  )

# Plot
ggplot(mutation_rate_by_smoking, aes(x = Hugo_Symbol, y = Percent, fill = Smoke.Status)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  labs(
    title = "Mutation Frequency in Key Genes by Smoking Status",
    x = "Gene",
    y = "Percentage of Patients with Mutation",
    fill = "Smoking Status"
  ) +
  theme_minimal(base_size = 14)
```

```{r}
# Nice correlation
smoking_status_df <- df %>% filter(Smoke.Status != "Unknown") %>%
  select(Smoke.Status, TMB..nonsynonymous.) %>%
  mutate(Smoke.Status = factor(Smoke.Status, levels = c("Smoker", "Nonsmoker")))
ggplot(smoking_status_df, aes(x = Smoke.Status, y = TMB..nonsynonymous., fill = Smoke.Status)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.3) +
  stat_compare_means(method = "wilcox.test", label.y = max(smoking_status_df$TMB..nonsynonymous., na.rm = TRUE) * 0.95) +
  labs(
    title = "Tumor Mutational Burden by Smoking Status",
    x = "Smoking Status",
    y = "Nonsynonymous Mutation Count (TMB)"
  ) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "none")
```

```{r}

# ONLY NON SMALL LUNG CANCER 

# Nice correlation
smoking_status_df <- df %>% filter(Smoke.Status != "Unknown" & `Cancer.Type` %in% c("Non Small Cell Lung Cancer")) %>%
  select(Smoke.Status, TMB..nonsynonymous.) %>%
  mutate(Smoke.Status = factor(Smoke.Status, levels = c("Smoker", "Nonsmoker")))

ggplot(smoking_status_df, aes(x = Smoke.Status, y = TMB..nonsynonymous., fill = Smoke.Status)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
  # geom_jitter(width = 0.2, size = 1, alpha = 0.3) +
  stat_compare_means(method = "wilcox.test", label.y = max(smoking_status_df$TMB..nonsynonymous., na.rm = TRUE) * 0.95) +
  labs(
    title = "Tumor Mutational Burden by Smoking Status",
    x = "Smoking Status",
    y = "Nonsynonymous Mutation Count (TMB)"
  ) +
    scale_fill_manual(
    values = c("Smoker" = "#d73027",    # אדום חזק - צבע אזהרה
               "Nonsmoker" = "#1a9850") # ירוק נעים
  ) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "none")
```

```{r}

# ONLY SMALL LUNG CANCER 

# Nice correlation
smoking_status_df <- df %>% filter(Smoke.Status != "Unknown" & `Cancer.Type` %in% c("Small Cell Lung Cancer")) %>%
  select(Smoke.Status, TMB..nonsynonymous.) %>%
  mutate(Smoke.Status = factor(Smoke.Status, levels = c("Smoker", "Nonsmoker")))
ggplot(smoking_status_df, aes(x = Smoke.Status, y = TMB..nonsynonymous., fill = Smoke.Status)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.3) +
  stat_compare_means(method = "wilcox.test", label.y = max(smoking_status_df$TMB..nonsynonymous., na.rm = TRUE) * 0.95) +
  labs(
    title = "Tumor Mutational Burden by Smoking Status",
    x = "Smoking Status",
    y = "Nonsynonymous Mutation Count (TMB)"
  ) +
  theme_minimal(base_size = 14) +
  theme(legend.position = "none")
```



```{r}
library(ggplot2)
library(dplyr)

# קריאה לקובץ
data <- read.csv("./somatic_alterations_from_tumors.csv")

# יצירת טבלה מסוכמת
summary_df <- data %>% filter(Smoke.Status != "Unknown") %>%
  group_by(`Cancer.Type`, `Smoke.Status`) %>%
  summarise(mean_TMB = mean(TMB..nonsynonymous., na.rm = TRUE)) %>%
  ungroup()

# ציור Heatmap
ggplot(summary_df, aes(x = `Smoke.Status`, y = `Cancer.Type`, fill = mean_TMB)) +
  geom_tile(color = "white") +
  scale_fill_viridis_c(name = "Mean TMB") +
  theme_minimal() +
  labs(title = "Mean TMB by Cancer Type and Smoking Status",
       x = "Smoke Status",
       y = "Cancer Type") +
  theme(axis.text.y = element_text(size = 8))




```


```{r}

table(data$Cancer.Type[data$Cancer.Type %in% c("Small Cell Lung Cancer", "Non Small Cell Lung Cancer")])


```



```{r}
# HEATMAP BY TMP - cancer type and chromosome

summary_df <- data %>%
  group_by(Cancer.Type, Chromosome) %>%
  summarise(TMB_count = n()) %>%
  filter(!is.na(Chromosome)) %>%
  ungroup()

# fill with 0 non existing combinations
summary_df <- summary_df %>%
  complete(Cancer.Type, Chromosome, fill = list(TMB_count = 0))

ordered_chr <- c(as.character(1:22), "X")
data$Chromosome <- factor(data$Chromosome, levels = ordered_chr)



# ציור heatmap
ggplot(summary_df, aes(x = Chromosome, y = Cancer.Type, fill = TMB_count)) +
  geom_tile(color = "white") +
  scale_fill_viridis_c(name = "TMB Count") +
  theme_minimal() +
  labs(title = "Heatmap of TMB Count by Cancer Type and Chromosome",
       x = "Chromosome",
       y = "Cancer Type") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.text.y = element_text(size = 8))

```


```{r}


library(ggplot2)
library(dplyr)

# קריאת הקובץ
data <- read.csv("somatic_alterations_from_tumors.csv")

# מציאת 20 הגנים הכי שכיחים
top_genes <- data %>%
  count(Hugo_Symbol, sort = TRUE) %>%
  slice_head(n = 20) %>%
  pull(Hugo_Symbol)

# סינון הנתונים רק לאותם גנים
filtered_data <- data %>%
  filter(Hugo_Symbol %in% top_genes)

# טבלה מסוכמת: כמה מופעים יש לכל סוג סרטן ולכל גן
summary_df <- filtered_data %>%
  group_by(Cancer.Type, Hugo_Symbol) %>%
  summarise(TMB_count = n()) %>%
  ungroup()

# ציור heatmap
ggplot(summary_df, aes(x = Hugo_Symbol, y = Cancer.Type, fill = TMB_count)) +
  geom_tile(color = "white") +
  scale_fill_viridis_c(name = "TMB Count") +
  theme_minimal() +
  labs(title = "TMB Count by Cancer Type and Top 20 Mutated Genes",
       x = "Gene (Hugo Symbol)",
       y = "Cancer Type")

```


```{r}

library(ggplot2)
library(dplyr)
library(tidyr)
library(forcats)
library(RColorBrewer)


# קריאת הקובץ
data <- read.csv("somatic_alterations_from_tumors.csv")

# סינון שורות עם כרומוזום חסר
data <- data %>% filter(!is.na(Chromosome))

# מיון כרומוזומים לפי סדר טבעי: 1–22 ואז X
ordered_chr <- c(as.character(1:22), "X")
data$Chromosome <- factor(data$Chromosome, levels = ordered_chr)

# טבלה מסוכמת: מספר מוטציות לכל שילוב
summary_df <- data %>%
  group_by(Cancer.Type, Chromosome) %>%
  summarise(TMB_count = n()) %>%
  filter(!is.na(Chromosome)) %>%
  ungroup()

# השלמת כל השילובים החסרים עם 0
summary_df <- summary_df %>%
  complete(Cancer.Type, Chromosome, fill = list(TMB_count = 0))

# מיון סוגי הסרטן לפי סכום כולל של TMB (מהגבוה לנמוך)
summary_df <- summary_df %>%
  group_by(Cancer.Type) %>%
  mutate(TMB_total = sum(TMB_count)) %>%
  ungroup() %>%
  mutate(Cancer.Type = fct_reorder(Cancer.Type, TMB_total, .desc = TRUE))

# 6. חישוב TMB הכולל לכל סוג סרטן ולמציאת הטופ 10
top10_cancers <- summary_df %>%
  group_by(Cancer.Type) %>%
  summarise(TMB_total = sum(TMB_count), .groups = "drop") %>%
  arrange(desc(TMB_total)) %>%
  slice_head(n = 12) %>%
  pull(Cancer.Type)

# 7. סינון לטופ 10 בלבד
summary_top10 <- summary_df %>%
  filter(Cancer.Type %in% top10_cancers) %>%
  # מיון Cancer.Type לפי TMB_total יורד
  group_by(Cancer.Type) %>%
  mutate(TMB_total = sum(TMB_count)) %>%
  ungroup() %>%
  mutate(Cancer.Type = fct_reorder(Cancer.Type, TMB_total, .desc = TRUE))


# ציור Heatmap
ggplot(summary_top10, aes(x = Chromosome, y = Cancer.Type, fill = TMB_count)) +
  geom_tile(color = "white") +
    # scale_fill_gradient2(name = "TMB Count",
    #                    # low = "green",
    #                    mid = "black",
    #                    high = "red",
    #                    midpoint = median(summary_top10$TMB_count)) +
    scale_fill_distiller(
    name = "TMB Count",
    palette = "OrRd",
    direction = 1,
    limits = c(0, max(summary_top10$TMB_count, na.rm = TRUE))
  ) +
  # scale_fill_viridis_c(name = "TMB Count") +
  theme_minimal() +
  labs(title = "TMB Count by Cancer Type and Chromosome",
       x = "Chromosome",
       y = "Cancer Type") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        axis.text.y = element_text(size = 8))

```



```{r}
# explanatory - with removing dups

library(dplyr)
library(ggplot2)
library(ggpubr)

# יצירת טבלה של חולים ייחודיים בלבד
smoking_status_unique <- df %>%
  filter(Smoke.Status %in% c("Smoker", "Nonsmoker"),
         Cancer.Type == "Non Small Cell Lung Cancer") %>%
  select(PATIENT_ID, Smoke.Status, TMB..nonsynonymous.) %>%
  distinct(PATIENT_ID, .keep_all = TRUE) %>%
  mutate(Smoke.Status = factor(Smoke.Status, levels = c("Smoker", "Nonsmoker")))

# ציור בוקספלאט
ggplot(smoking_status_unique, aes(x = Smoke.Status, y = TMB..nonsynonymous., fill = Smoke.Status)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7) +
  stat_compare_means(
    method = "wilcox.test",
    label.y = max(smoking_status_unique$TMB..nonsynonymous., na.rm = TRUE) * 0.95
  ) +
  scale_fill_manual(
    values = c("Smoker" = "#d73027", "Nonsmoker" = "#1a9850")
  ) +
  labs(
    title = "Tumor Mutational Burden by Smoking Status (Unique Patients)",
    x = "Smoking Status",
    y = "Nonsynonymous Mutation Count (TMB)"
  ) +
  theme_minimal(base_size = 14) +
  coord_cartesian(ylim = c(0, 1.5))+
  theme(legend.position = "none")


smokers <- smoking_status_unique %>% filter(Smoke.Status == "Smoker") %>% pull(TMB..nonsynonymous.)
nonsmokers <- smoking_status_unique %>% filter(Smoke.Status == "Nonsmoker") %>% pull(TMB..nonsynonymous.)

wilcox.test(smokers, nonsmokers)

```
